# Практична робота №6
## Тема: Описова статистика: міри положення і розсіювання, describe, порівняння груп
**Дисципліна:** Аналіз даних та математичне моделювання  
**Студент:** Хрустовський Павло  
**Група:** ІТ-42  
**Варіант:** 2 (Місто: Львів, Регіон: Захід)  

---

### Мета роботи:
Одним викликом `.describe()` отримати всі основні міри положення й розсіювання з лекції 6, зіставити їх із формулами, а потім порівняти описову статистику між умовними регіонами — без `groupby()`, лише булевою фільтрацією.


## Завдання 1. describe() на власному варіанті

1. Побудувати `pd.Series` (або один стовпець DataFrame) із 12 значень середньомісячної відносної вологості повітря (%) для свого міста (Львів, Варіант 2).
2. Викликати метод `.describe()` та письмово охарактеризувати, яка міра положення чи розсіювання з лекції 6 відповідає кожному з рядків: `mean`, `std`, `min`, `25%`, `50%`, `75%`, `max`.


In [1]:
import pandas as pd
import numpy as np

# Список назв місяців
months = ["Січ", "Лют", "Бер", "Кві", "Тра", "Чер", "Лип", "Сер", "Вер", "Жов", "Лис", "Гру"]

# Дані середньомісячної вологості для Львова (Варіант 2)
lviv_data = [85, 83, 80, 73, 68, 65, 63, 65, 70, 77, 82, 86]

# Побудова pd.Series
lviv_humidity = pd.Series(lviv_data, index=months, name="Вологість_Львів")

print("Середньомісячна відносна вологість у м. Львів (%):")
print(lviv_humidity)
print("\n" + "=" * 45 + "\n")

# Виклик .describe() на власному варіанті
lviv_describe = lviv_humidity.describe()
print("Результат виклику .describe():")
print(lviv_describe)


Середньомісячна відносна вологість у м. Львів (%):
Січ    85
Лют    83
Бер    80
Кві    73
Тра    68
Чер    65
Лип    63
Сер    65
Вер    70
Жов    77
Лис    82
Гру    86
Name: Вологість_Львів, dtype: int64


Результат виклику .describe():
count    12.000000
mean     74.750000
std       8.443664
min      63.000000
25%      67.250000
50%      75.000000
75%      82.250000
max      86.000000
Name: Вологість_Львів, dtype: float64


### Письмовий розбір показників виводу `.describe()`:

Кожен рядок у виводі методу `.describe()` відповідає певній статистичній мірі положення чи розсіювання з лекційного матеріалу:

1. **`count` (12.000000)** — **Кількість спостережень ($N$)**: загальний обсяг вибірки (12 календарних місяців року).
2. **`mean` (74.750000 %)** — **Вибіркове середнє арифметичне ($\\bar{x}$)**: ключова параметрична **міра центрального положення**. Розраховується як сума всіх значень вологості, поділена на їхню кількість:
   $$\\bar{x} = \\frac{1}{N} \\sum_{i=1}^{N} x_i$$
3. **`std` (8.443664 %)** — **Вибіркове стандартне відхилення ($s$)**: основна параметрична **міра розсіювання**. За замовчуванням у `pandas` обчислюється з поправкою Бесселя ($\\text{ddof}=1$), характеризуючи середній розкид помісячних показників навколо середнього значення:
   $$s = \\sqrt{\\frac{1}{N-1} \\sum_{i=1}^{N} (x_i - \\bar{x})^2}$$
4. **`min` (63.000000 %)** — **Мінімум вибірки**: крайня порядкова статистика, **міра положення** (найсухіший місяць — липень, $63\\%$). Разом із максимумом служить базою для обчислення розмаху.
5. **`25%` (67.250000 %)** — **Перший квартиль ($Q_1$, або 25-й процентиль)**: **міра положення**. Показує, що у чверті місяців року ($25\\%$) вологість повітря не перевищувала $67.25\\%$.
6. **`50%` (75.000000 %)** — **Другий квартиль / Медіана ($Q_2$, $\\text{Med}$)**: центральна робастна **міра положення**. Ділить впорядковані значення навпіл: у половині місяців вологість була $\\le 75\\%$, а в іншій половині — $\\ge 75\\%$.
7. **`75%` (82.250000 %)** — **Третій квартиль ($Q_3$, або 75-й процентиль)**: **міра положення**. Показує, що у $75\\%$ місяців вологість не перевищувала $82.25\\%$. Разом із $Q_1$ формує інтерквартильний розмах ($IQR$).
8. **`max` (86.000000 %)** — **Максимум вибірки**: крайня порядкова статистика, **міра положення** (найбільш вологий місяць — грудень, $86\\%$).


## Завдання 2. Ручне обчислення і перевірка

1. Обчислити окремо прямими методами без `describe()`:
   - розмах ($\\text{Range} = \\text{max} - \\text{min}$);
   - дисперсію ($\\text{ddof}=1$ і окремо $\\text{ddof}=0$, з порівнянням різниці);
   - стандартне відхилення ($s$);
   - інтерквартильний розмах ($IQR = Q_3 - Q_1$).
2. Переконатися у точному збігу значень квартилів $25\\%$ і $75\\%$ з виводом `.describe()`.
3. Обчислити коефіцієнт варіації ($CV = \\frac{s}{\\bar{x}} \\times 100\\%$).


In [2]:
# 1. Розмах (Range)
range_val = lviv_humidity.max() - lviv_humidity.min()

# 2. Дисперсія: з поправкою Бесселя (ddof=1) та без (ddof=0)
var_ddof1 = lviv_humidity.var(ddof=1)
var_ddof0 = lviv_humidity.var(ddof=0)
var_diff = var_ddof1 - var_ddof0
var_ratio = var_ddof1 / var_ddof0

# 3. Стандартне відхилення (std)
std_ddof1 = lviv_humidity.std(ddof=1)
std_ddof0 = lviv_humidity.std(ddof=0)

# 4. Квартилі та інтерквартильний розмах (IQR)
q1 = lviv_humidity.quantile(0.25)
q3 = lviv_humidity.quantile(0.75)
iqr = q3 - q1

# 5. Коефіцієнт варіації (CV)
mean_val = lviv_humidity.mean()
cv = (std_ddof1 / mean_val) * 100

print(f"Розмах (Range = max - min):        {range_val:.2f}%")
print(f"Дисперсія незсунена (ddof=1):      {var_ddof1:.4f}")
print(f"Дисперсія генеральна (ddof=0):     {var_ddof0:.4f}")
print(f"Абсолютна різниця дисперсій:       {var_diff:.4f}")
print(f"Відношення var(ddof=1)/var(ddof=0):{var_ratio:.4f} (теоретичне N/(N-1) = {12/11:.4f})")
print(f"Стандартне відхилення (ddof=1):   {std_ddof1:.4f}%")
print(f"Стандартне відхилення (ddof=0):   {std_ddof0:.4f}%")
print(f"Перший квартиль Q1 (25%):          {q1:.2f}% (збіг з describe: {q1 == lviv_describe['25%']})")
print(f"Третій квартиль Q3 (75%):          {q3:.2f}% (збіг з describe: {q3 == lviv_describe['75%']})")
print(f"Інтерквартильний розмах (IQR):     {iqr:.2f}%")
print(f"Коефіцієнт варіації (CV):          {cv:.2f}%")


Розмах (Range = max - min):        23.00%
Дисперсія незсунена (ddof=1):      71.2955
Дисперсія генеральна (ddof=0):     65.3542
Абсолютна різниця дисперсій:       5.9413
Відношення var(ddof=1)/var(ddof=0):1.0909 (теоретичне N/(N-1) = 1.0909)
Стандартне відхилення (ddof=1):   8.4437%
Стандартне відхилення (ddof=0):   8.0842%
Перший квартиль Q1 (25%):          67.25% (збіг з describe: True)
Третій квартиль Q3 (75%):          82.25% (збіг з describe: True)
Інтерквартильний розмах (IQR):     15.00%
Коефіцієнт варіації (CV):          11.30%


### Аналіз результатів ручного розрахунку:

1. **Порівняння дисперсій при $\\text{ddof}=1$ та $\\text{ddof}=0$:**
   - Значення вибіркової незсуненої дисперсії ($s^2 = 71.2955$) помітно перевищує оцінку генеральної дисперсії без поправки ($\\sigma^2 = 65.3542$).
   - Їхнє співвідношення дорівнює точно $\\frac{N}{N - 1} = \\frac{12}{11} \\approx 1.0909$ (різниця складає близько $9.09\\%$).
   - Оскільки вибірка є невеликою ($N = 12$), використання вибіркового середнього $\\bar{x}$ замість істинного математичного сподівання суттєво занижує варіацію. Поправка Бесселя ($\\text{ddof}=1$) усуває цей зсув і дає об\'єктивну незсунену оцінку.
2. **Точність розрахунку квартилів:**
   - Значення $Q_1 = 67.25\\%$ та $Q_3 = 82.25\\%$ повністю ідентичні рядкам `25%` та `75%` у `.describe()`.
   - Інтерквартильний розмах $IQR = 15.00\\%$ охоплює центральні $50\\%$ річного розподілу вологості.
3. **Оцінка коефіцієнта варіації ($CV$):**
   - $CV = 11.30\\%$. Згідно зі статистичною шкалою оцінки варіації ($CV < 10\\%$ — незначна, $10\\% \\le CV \\le 20\\%$ — середня/помірна, $CV > 20\\%$ — висока), вологість повітря у Львові має **помірну варіативність**, а сукупність спостережень є однорідною.


## Завдання 3. Порівняння регіонів

1. Побудувати повний `DataFrame` з усіх 10 міст та 12 місяців за наданою таблицею варіантів.
2. Додати стовпець `Середня_річна_вологість` як середнє значення по 12 місяцях кожного міста.
3. За допомогою **булевої фільтрації** (без `groupby()`) вибрати свій регіон ("Захід") та вивести `.describe()`.
4. За допомогою булевої фільтрації вибрати інший регіон ("Схід") та вивести `.describe()`.
5. Письмово порівняти їхній розкид ($std$) та коефіцієнт варіації ($CV$).


In [3]:
# Таблиця всіх варіантів міст
table_data = [
    {"Варіант": 1, "Місто": "Київ", "Регіон": "Центр", "Січ": 84, "Лют": 82, "Бер": 78, "Кві": 70, "Тра": 65, "Чер": 63, "Лип": 61, "Сер": 63, "Вер": 68, "Жов": 75, "Лис": 81, "Гру": 85},
    {"Варіант": 2, "Місто": "Львів", "Регіон": "Захід", "Січ": 85, "Лют": 83, "Бер": 80, "Кві": 73, "Тра": 68, "Чер": 65, "Лип": 63, "Сер": 65, "Вер": 70, "Жов": 77, "Лис": 82, "Гру": 86},
    {"Варіант": 3, "Місто": "Одеса", "Регіон": "Південь", "Січ": 82, "Лют": 80, "Бер": 78, "Кві": 74, "Тра": 71, "Чер": 69, "Лип": 67, "Сер": 68, "Вер": 72, "Жов": 76, "Лис": 80, "Гру": 83},
    {"Варіант": 4, "Місто": "Харків", "Регіон": "Схід", "Січ": 83, "Лют": 80, "Бер": 75, "Кві": 68, "Тра": 62, "Чер": 60, "Лип": 58, "Сер": 60, "Вер": 66, "Жов": 73, "Лис": 79, "Гру": 84},
    {"Варіант": 5, "Місто": "Дніпро", "Регіон": "Схід", "Січ": 82, "Лют": 79, "Бер": 74, "Кві": 67, "Тра": 61, "Чер": 59, "Лип": 57, "Сер": 59, "Вер": 65, "Жов": 72, "Лис": 78, "Гру": 83},
    {"Варіант": 6, "Місто": "Чернігів", "Регіон": "Центр", "Січ": 85, "Лют": 83, "Бер": 79, "Кві": 71, "Тра": 66, "Чер": 63, "Лип": 61, "Сер": 63, "Вер": 69, "Жов": 76, "Лис": 82, "Гру": 86},
    {"Варіант": 7, "Місто": "Івано-Франківськ", "Регіон": "Захід", "Січ": 86, "Лют": 84, "Бер": 81, "Кві": 75, "Тра": 70, "Чер": 67, "Лип": 65, "Сер": 67, "Вер": 72, "Жов": 78, "Лис": 83, "Гру": 87},
    {"Варіант": 8, "Місто": "Полтава", "Регіон": "Центр", "Січ": 84, "Лют": 81, "Бер": 76, "Кві": 69, "Тра": 63, "Чер": 61, "Лип": 59, "Сер": 61, "Вер": 67, "Жов": 74, "Лис": 80, "Гру": 85},
    {"Варіант": 9, "Місто": "Суми", "Регіон": "Схід", "Січ": 85, "Лют": 82, "Бер": 77, "Кві": 70, "Тра": 64, "Чер": 62, "Лип": 60, "Сер": 62, "Вер": 68, "Жов": 75, "Лис": 81, "Гру": 86},
    {"Варіант": 10, "Місто": "Ужгород", "Регіон": "Захід", "Січ": 84, "Лют": 82, "Бер": 79, "Кві": 73, "Тра": 68, "Чер": 66, "Лип": 64, "Сер": 66, "Вер": 71, "Жов": 77, "Лис": 82, "Гру": 85},
]

df_cities = pd.DataFrame(table_data)

# Стовпець із середньою вологістю за рік для кожного міста
df_cities["Середня_річна_вологість"] = df_cities[months].mean(axis=1)

print("Загальна таблиця міст та їх середньорічної вологості:")
print(df_cities[["Варіант", "Місто", "Регіон", "Середня_річна_вологість"]])
print("\n" + "=" * 55 + "\n")

# Булева фільтрація для регіону "Захід"
west_subset = df_cities[df_cities["Регіон"] == "Захід"]["Середня_річна_вологість"]
west_desc = west_subset.describe()
west_cv = (west_subset.std() / west_subset.mean()) * 100

print("Описова статистика середньорічної вологості: Регіон 'Захід'")
print(west_desc)
print(f"Коефіцієнт варіації (CV) для регіону 'Захід': {west_cv:.2f}%\n")

# Булева фільтрація для регіону "Схід"
east_subset = df_cities[df_cities["Регіон"] == "Схід"]["Середня_річна_вологість"]
east_desc = east_subset.describe()
east_cv = (east_subset.std() / east_subset.mean()) * 100

print("Описова статистика середньорічної вологості: Регіон 'Схід'")
print(east_desc)
print(f"Коефіцієнт варіації (CV) для регіону 'Схід': {east_cv:.2f}%")


Загальна таблиця міст та їх середньорічної вологості:
   Варіант             Місто   Регіон  Середня_річна_вологість
0        1              Київ    Центр                72.916667
1        2             Львів    Захід                74.750000
2        3             Одеса  Південь                75.000000
3        4            Харків     Схід                70.666667
4        5            Дніпро     Схід                69.666667
5        6          Чернігів    Центр                73.666667
6        7  Івано-Франківськ    Захід                76.250000
7        8           Полтава    Центр                71.666667
8        9              Суми     Схід                72.666667
9       10           Ужгород    Захід                74.750000


Описова статистика середньорічної вологості: Регіон 'Захід'
count     3.000000
mean     75.250000
std       0.866025
min      74.750000
25%      74.750000
50%      74.750000
75%      75.500000
max      76.250000
Name: Середня_річна_вологість, dtype: f

### Письмове порівняння регіонів «Захід» та «Схід»:

| Параметр | Регіон «Захід» (Львів, Івано-Франківськ, Ужгород) | Регіон «Схід» (Харків, Дніпро, Суми) |
| :--- | :---: | :---: |
| **Кількість міст ($N$)** | 3 | 3 |
| **Середнє ($\\bar{x}$)** | **75.25 %** | **71.00 %** |
| **Стандартне відхилення ($std$)** | **0.8660 %** | **1.5275 %** |
| **Мінімум / Максимум** | 74.75 % / 76.25 % | 69.67 % / 72.67 % |
| **Розмах ($Range$)** | 1.50 % | 3.00 % |
| **Коефіцієнт варіації ($CV$)** | **1.15 %** | **2.15 %** |

**Висновки:**
1. **Порівняння розкиду ($std$):**
   - У регіону «Схід» стандартне відхилення середньорічної вологості становить $std = 1.53\\%$, тоді як у регіону «Захід» воно дорівнює лише $std = 0.87\\%$. Розмах на Сході становить $3.00\\%$, що вдвічі ширше за розмах на Заході ($1.50\\%$).
   - Таким чином, середньорічна вологість між містами Сходу варіюється відчутно сильніше, ніж між містами Заходу.
2. **Узгодженість із коефіцієнтом варіації ($CV$):**
   - Розрахунок коефіцієнта варіації повністю підтверджує цей висновок: у регіону «Схід» $CV = 2.15\\%$, що майже вдвічі більше за $CV = 1.15\\%$ у регіону «Захід».
   - Обидва регіони є високооднорідними ($CV < 5\\%$), проте кліматичні умови вологості в межах Західного регіону є суттєво стабільнішими та ближчими між містами, ніж у Східному регіоні.


## Завдання 4. Форма розподілу власних даних

1. За 12 місячними значеннями вологості міста Львів визначити, чи розподіл є симетричним, чи скошеним (порівняти `mean` та `50%` медіану з Завдання 1).
2. Письмово обґрунтувати, яка міра положення (середнє арифметичне чи медіана) є більш репрезентативною для наявних даних.


In [4]:
mean_lviv = lviv_humidity.mean()
median_lviv = lviv_humidity.median()
diff = mean_lviv - median_lviv
rel_diff = (diff / mean_lviv) * 100

# Перевірка симетрії квартилів навколо медіани
dist_q1_med = median_lviv - q1
dist_med_q3 = q3 - median_lviv

print(f"Вибіркове середнє (mean):          {mean_lviv:.2f}%")
print(f"Медіана (50%):                     {median_lviv:.2f}%")
print(f"Абсолютна різниця (mean - median): {diff:.2f} п.п.")
print(f"Відносна різниця:                  {rel_diff:.2f}%")
print("-" * 45)
print(f"Відстань від Q1 до медіани:        {dist_q1_med:.2f}%")
print(f"Відстань від медіани до Q3:        {dist_med_q3:.2f}%")


Вибіркове середнє (mean):          74.75%
Медіана (50%):                     75.00%
Абсолютна різниця (mean - median): -0.25 п.п.
Відносна різниця:                  -0.33%
---------------------------------------------
Відстань від Q1 до медіани:        7.75%
Відстань від медіани до Q3:        7.25%


### Висновки щодо форми розподілу та вибору міри положення:

1. **Оцінка форми розподілу:**
   - Середнє значення становить $\\bar{x} = 74.75\\%$, а медіана — $\\text{Med} = 75.00\\%$.
   - Абсолютна різниця між ними складає лише $-0.25$ п.п., що становить мізерні $0.33\\%$ від середнього.
   - Відстані від першого квартиля до медіани ($7.75\\%$) та від медіани до третього квартиля ($7.25\\%$) майже ідентичні (різниця всього $0.50\\%$).
   - Це однозначно свідчить про те, що розподіл середньомісячної вологості у Львові є **практично симетричним**, без вираженої правої чи лівої скошеності. Значення рівномірно спадають від центру до крайніх точок.
2. **Обґрунтування вибору міри положення:**
   - Оскільки розподіл симетричний і у вибірці немає викидів чи аномальних перекосів, **найбільш репрезентативною мірою положення є середнє арифметичне ($\\bar{x} = 74.75\\%$)**.
   - **Чому саме так:** Середнє арифметичне є найбільш ефективною незсуненою оцінкою математичного сподівання для симетричних розподілів (має найменшу дисперсію серед усіх незсунених оцінок). Воно використовує точні чисельні значення всіх 12 місяців без втрати інформації, на відміну від медіани, яка спирається лише на ранги значень.
   - Збіг медіани ($75.00\\%$) із середнім підтверджує надійність середнього арифметичного.


## Завдання 5. Порівняння варіативності двох міст

1. Обрати будь-яке місто з таблиці варіантів, що належить до іншого регіону (Харків, регіон "Схід").
2. Обчислити для нього коефіцієнт варіації ($CV$) і порівняти зі значенням для Львова (Завдання 2).
3. Письмово пояснити, чому саме коефіцієнт варіації, а не сирий $std$, тут є коректним інструментом порівняння, та чи применшує відносна близькість середніх потребу в $CV$.


In [5]:
# Дані для міста Харків (Варіант 4, регіон "Схід")
kharkiv_data = df_cities[df_cities["Місто"] == "Харків"][months].iloc[0]

# Розрахунок параметрів для Харкова
kharkiv_mean = kharkiv_data.mean()
kharkiv_std = kharkiv_data.std(ddof=1)
kharkiv_cv = (kharkiv_std / kharkiv_mean) * 100

# Порівняльна таблиця двох міст
city_comparison = pd.DataFrame({
    "Місто": ["Львів", "Харків"],
    "Регіон": ["Захід", "Схід"],
    "Середнє (mean), %": [mean_val, kharkiv_mean],
    "Стандартне відхилення (std), %": [std_ddof1, kharkiv_std],
    "Коефіцієнт варіації (CV), %": [cv, kharkiv_cv]
})

print("Порівняння варіативності вологості міст Львів та Харків:")
print(city_comparison.to_string(index=False))


Порівняння варіативності вологості міст Львів та Харків:
 Місто Регіон  Середнє (mean), %  Стандартне відхилення (std), %  Коефіцієнт варіації (CV), %
 Львів  Захід          74.750000                        8.443664                    11.295871
Харків   Схід          70.666667                        9.547330                    13.510373


### Письмове пояснення результатів порівняння міст:

1. **Аналіз отриманих показників:**
   - Львів (Захід): $\\bar{x} = 74.75\\%$, $std = 8.44\\%$, $CV = 11.30\\%$.
   - Харків (Схід): $\\bar{x} = 70.67\\%$, $std = 9.55\\%$, $CV = 13.51\\%$.
   - В обох вимірах вологість у Харкові є більш мінливою протягом року: абсолютний розкид $std$ вищий на $1.11$ п.п., а відносний розкид $CV$ — на $2.21$ процентного пункту. Це зумовлено сильнішою континентальністю східного клімату (значно сухіше літо з вологістю до $58\\%$ при високій зимовій вологості до $84\\%$).

2. **Чому саме коефіцієнт варіації ($CV$), а не сирий $std$, є коректним інструментом:**
   - Стандартне відхилення ($std$) вимірюється в абсолютних одиницях вихідної шкали (процентні пункти) і фіксує лише відхилення від власного середнього.
   - Проте мінливість процесу залежить від його масштабу та базового рівня. Коефіцієнт варіації $CV = \\frac{s}{\\bar{x}} \\times 100\\%$ нормує абсолютне розсіювання на середнє значення, усуваючи вплив різниці масштабу. Це перетворює розсіювання на безрозмірну відносну величину, що робить порівняння статистично строгим.

3. **Чи применшує близькість середніх потребу в $CV$:**
   - **Ні, не применшує.** Хоча середні значення вологості міст відносно близькі ($74.75\\%$ та $70.67\\%$), між ними все одно існує різниця у понад $4$ процентні пункти.
   - Використання $CV$ є методологічним стандартом математичної статистики, оскільки воно гарантує коректність оцінки незалежно від того, наскільки близькими виявилися середні. Крім того, саме коефіцієнт варіації дозволяє класифікувати ступінь однорідності вибірки за загальноприйнятими критеріями.


## Відповіді на контрольні / теоретичні запитання

---

### 1. Чи можете пояснити, яка міра розсіювання з лекції стоїть за кожним із рядків `std`, `25%/75%` і `min/max` у виводі `.describe()`?

У стандартному виводі методу `.describe()` містяться як безпосередні міри розсіювання, так і базові компоненти для їхнього обчислення:

1. **`min` та `max` (Розмах вибірки, $\\text{Range}$):**
   - Самі значення `min` і `max` є крайніми порядковими статистиками (мірами положення), проте їхня різниця:
     $$\\text{Range} = \\text{max} - \\text{min}$$
     утворює **розмах вибірки** — найпростішу міру розсіювання. Вона відображає повний діапазон варіювання ознаки ($100\\%$ спостережень). Недоліком розмаху є його крайня нестійкість до поодиноких викидів.
2. **`25%` та `75%` (Інтерквартильний розмах, $IQR$):**
   - Значення `25%` ($Q_1$) та `75%` ($Q_3$) представляють перший і третій квартилі розподілу (міри положення). Їхня різниця:
     $$IQR = Q_3 - Q_1$$
     утворює **інтерквартильний розмах**. Це робастна міра розсіювання, яка описує діапазон центральних $50\\%$ вибірки, повністю відсікаючи по $25\\%$ екстремальних значень з обох країв. Вона не реагує на поодинокі викиди й є базою для виявлення аномалій (правило Тьюкі $1.5 \\times IQR$).
3. **`std` (Вибіркове стандартне відхилення, $s$):**
   - Рядок `std` відображає **вибіркове середньоквадратичне (стандартне) відхилення** з поправкою Бесселя ($\\text{ddof}=1$ за замовчуванням):
     $$s = \\sqrt{\\frac{1}{N-1} \\sum_{i=1}^{N} (x_i - \\bar{x})^2}$$
   - Це фундаментальна параметрична міра розсіювання, яка вимірює середнє відхилення значень від середнього арифметичного в одиницях вимірювання ознаки. Вона використовує інформацію з усіх точок вибірки, але чутлива до викидів.

---

### 2. Чи розумієте, чому заповнення пропуску середнім штучно зменшує `std`, і чи можете показати механіку цього на конкретних числах?

**Механіка явища:**
- Вибіркова дисперсія обчислюється як сума квадратів відхилень спостережень від середнього, поділена на кількість ступенів вільності ($N-1$):
  $$s^2 = \\frac{\\sum_{i=1}^{N} (x_i - \\bar{x})^2}{N - 1}$$
- Коли пропущене спостереження замінюється вибірковим середнім значенням ($x_{\\text{new}} = \\bar{x}$), його відхилення від середнього стає строго рівним нулю:
  $$(x_{\\text{new}} - \\bar{x})^2 = (\\bar{x} - \\bar{x})^2 = 0$$
- В результаті:
  - Сума квадратів відхилень у чисельнику **залишається незмінною** (додається $0$).
  - Знаменник (кількість спостережень та ступенів вільності) **збільшується на 1**.
  - Ділення тієї самої суми на більше число обов\'язково зменшує дисперсію $s^2$ і стандартне відхилення $s$. Розподіл штучно концентрується навколо середнього, що створює оманливу ілюзію меншої мінливості даних.

**Числовий приклад:**
1. Нехай маємо початковий ряд із 3 значень: $X = [10, 20, 30]$.
   - Середнє: $\\bar{x} = \\frac{10 + 20 + 30}{3} = 20$.
   - Сума квадратів відхилень:
     $$SS = (10 - 20)^2 + (20 - 20)^2 + (30 - 20)^2 = (-10)^2 + 0^2 + 10^2 = 100 + 0 + 100 = 200$$
   - Дисперсія: $s^2 = \\frac{200}{3 - 1} = \\frac{200}{2} = 100$.
   - Стандартне відхилення: $s = \\sqrt{100} = 10.0$.
2. Заповнюємо пропущене 4-те значення середнім ($20$): $X_{\\text{filled}} = [10, 20, 30, 20]$.
   - Нове середнє: $\\bar{x}_{\\text{new}} = \\frac{10 + 20 + 30 + 20}{4} = 20$ (не змінилося).
   - Нова сума квадратів відхилень:
     $$SS_{\\text{new}} = 200 + (20 - 20)^2 = 200 + 0 = 200$$ (чисельник залишився 200).
   - Нова дисперсія: $s_{\\text{new}}^2 = \\frac{200}{4 - 1} = \\frac{200}{3} \\approx 66.67$.
   - Нове стандартне відхилення: $s_{\\text{new}} = \\sqrt{66.67} \\approx 8.16$.
- **Висновок:** $std$ штучно знизилося з $10.0$ до $8.16$ (на $\\approx 18.4\\%$) виключно через операцію заповнення пропуску середнім.

---

### 3. Чи можете пояснити, чому порівнювати абсолютні значення `std` двох груп чи регіонів різного масштабу без коефіцієнта варіації може привести до хибного висновку?

**Пояснення:**
- Абсолютне стандартне відхилення ($std$) прямо залежить від одиниць вимірювання та середнього рівня досліджуваного показника. Чим більша абсолютна шкала ознаки, тим більшим зазвичай є і його $std$.
- Якщо порівнювати дві групи різного масштабу за сирим $std$, більша група практично завжди видаватиметься «більш варіативною», хоча відносно власного рівня вона може бути набагато стабільнішою.
- **Приклад:**
  - Розглянемо два активи:
    - Актив А (великий індекс): ціна $10\\,000$ грн, $std = 200$ грн.
    - Актив Б (невелика акція): ціна $100$ грн, $std = 10$ грн.
  - Порівняння за $std$: $std_A = 200 > std_B = 10$. Здається, що актив А значно більш ризикований і нестійкий.
  - Розрахунок коефіцієнта варіації:
    $$CV_A = \\frac{200}{10\\,000} \\times 100\\% = 2\\%$$
    $$CV_B = \\frac{10}{100} \\times 100\\% = 10\\%$$
  - **Висновок:** Актив Б у 5 разів більш волатильний відносно свого рівня ціни ($CV = 10\\%$ проти $2\\%$). Порівняння за сирим $std$ дало б прямо протилежний і хибний результат.

---

### 4. Судячи з власного розподілу (Завдання 4), чи він симетричний, і яка міра положення репрезентативніша саме для ваших даних — і чому?

**Відповідь для Варіанта 2 (Львів):**
1. **Симетричність розподілу:**
   - Середнє арифметичне: $\\bar{x} = 74.75\\%$.
   - Медіана (50%): $\\text{Med} = 75.00\\%$.
   - Різниця між ними становить лише $-0.25$ п.п. (менше $0.33\\%$), що практично дорівнює нулю.
   - Відстані від квартилів до медіани: $\\text{Med} - Q_1 = 7.75\\%$ та $Q_3 - \\text{Med} = 7.25\\%$. Вони відрізняються лише на $0.50\\%$.
   - Отже, розподіл місячної вологості у Львові є **майже ідеально симетричним**, без перекосів.
2. **Репрезентативність міри положення:**
   - Для наявних даних **середнє арифметичне ($\\bar{x} = 74.75\\%$) є найбільш репрезентативною мірою центрального положення**.
   - **Чому:**
     - У симетричних розподілах без виражених викидів вибіркове середнє є найкращою незсуненою оцінкою центру розподілу з мінімальною дисперсією.
     - Середнє враховує точні числові значення кожного з 12 місяців без винятку.
     - Медіана ($75.00\\%$) підтверджує цей результат, показуючи практично ту саму точку рівноваги.
